lib/rules/no-array-constructor.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * @fileoverview disallow generic `Array` constructors
+ * @author Jed Fox
+ * @author Matt DuVall <http://www.mattduvall.com/>
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow generic `Array` constructors",
+            category: "Stylistic Issues",
+            recommended: false
+        },
+        fixable: "code",
+        schema: []
+    },
+
+    create(context) {
+        /**
+         * Disallow construction of dense arrays using the Array constructor
+         * @param {ASTNode} node node to evaluate
+         * @returns {void}
+         * @private
+         */
+        function check(node) {
+            if (
```

Comment:
This is probably a long shot, but I'm going to say it anyway... 

The only valid constructors should be `new Array(number)` and `new Array(identifier)`, but the code allows `new Array('hello')`, `new Array(true)`, etc...

I know we don't have type information available at runtime to validate identifiers (e.g. `new Array(identifier)`, nothing we can do about them without complicating things), but I still think we can validate literals 😃, perhaps a typeof node.arguments[0].value === "number" when node.arguments[0].type === "Literal"?

Thoughts?

Reply:
It seems like doing `new Array(<anything that’s not a number>)` will just return an array with that parameter as the first element.

Reply:
That is correct. But isn't creating empty arrays with the given length what we are trying to validate? 
> Two exceptions are when the Array constructor is used to intentionally create sparse arrays of a specified size by giving the constructor a single numeric argument

Reply:
I’d open this as an issue on eslint/eslint.